src/azul/terraform.py

```diff
@@ -721,6 +722,29 @@ def tf_config(self, app_name):
                 key.replace('.', '_'): value
                 for key, value in bucket_notifications.items()
             }
+            # To prevent a race condition by Terraform, we make the bucket
+            # notifications depend on the related aws_lambda_permission and
+            # aws_lambda_function resources.
+            for notification in resources['aws_s3_bucket_notification'].values():
+                lambda_arns = [
+                    lambda_function['lambda_function_arn']
+                    for lambda_function in notification['lambda_function']
+                ]
+                lambda_resources = []
+                permission_resources = []
+                for key, value in resources['aws_lambda_permission'].items():
+                    lambda_arn = value['function_name']
+                    if lambda_arn in lambda_arns:
+                        match = re.fullmatch(r'\${(.*)\.arn}', lambda_arn)
```

Comment:
Please check if there isn't already existing functionality to parse a TF reference.